src/main/java/org/kitteh/irc/client/library/feature/SimpleDefaultingOutboundMessageMap.java

```diff
@@ -0,0 +1,144 @@
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.feature;
+
+import org.kitteh.irc.client.library.util.Sanity;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SimpleDefaultingOutboundMessageMap implements DefaultingOutboundMessageMap {
+    protected final Map<DefaultingOutboundMessage, String> defaults = new HashMap<>();
+    @Nullable protected final String defaultString;
+
+    /**
+     * Creates a SimpleDefaultingOutboundMessageMap object with a custom string to default
+     * to when an DefaultingOutboundMessage key is not set.
+     *
+     * @param defaultString a default string to use for when a key is not set
+     * @throws IllegalArgumentException Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     */
+    public SimpleDefaultingOutboundMessageMap(@Nullable String defaultString) {
+        defaultString = this.emptyStringCheck(defaultString);
+
+        this.defaultString = defaultString;
+    }
+
+    /**
+     * Creates a SimpleDefaultingOutboundMessageMap object with predefined string to default
+     * to when an DefaultingOutboundMessage key is not set.
+     */
+    public SimpleDefaultingOutboundMessageMap() {
+        this.defaultString = null;
+    }
+
+    /**
+     * Sets a default message for the DefaultingOutboundMessage.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is
+     * provided for this DefaultingOutboundMessage
+     * @return Returns this to permit chaining
+     * @throws IllegalArgumentException Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     */
+    public SimpleDefaultingOutboundMessageMap setDefault(@Nonnull DefaultingOutboundMessage key, @Nullable String defaultString) {
+        Sanity.nullCheck(key, "DefaultingOutboundMessage key must not be null");
+        defaultString = this.emptyStringCheck(defaultString);
+
+        this.defaults.put(key, defaultString);
```

Comment:
Avoid re-assigning the `defaultString` variable:
`this.defaults.put(key, this.emptyStringCheck(defaultString));`